### DISCUSSION TITLE
Trying to use custom fields with the inline plugin but I get a NameError

### DISCUSSION THREAD

**Discussion by vazaha-nl** [beetbox/beets#5375] [Q&A]
I am trying to use custom fields provided by another plugin with the inline plugin, but it does not seem to work. The standard fields are available in python but custom fields provided by other plugins give a python NameError.

Context: I use the [beets-artistcountry](https://github.com/agrausem/beets-artistcountry) plugin to get the country of the artist. It is available in the `$artist_country` variable and I can use it in my `paths` definition, like this, and it works:

```yaml
paths:
  albumtype:album: Modern/$artist_country/$albumartist/$album%aunique{}/$albumartist - $album - $track. $title
```

But I want to get one step deeper and retrieve the country name, and the continent for that artist_coutry. I am able to do that using the inline plugin and a python snippet. I got it working for the default `country` field but that is the country of release, not the artist country. So I want to do the same with `artist_country`.

However, `artist_country` seems not to be accessible in python using the inline plugin. I get a ` NameError: name 'artist_country' is not defined` error.

Relevant parts from my config:

```yaml
plugins:
  - artistcountry
  - inline

item_fields:
  country_name: |
    import pycountry_convert
    country_name = pycountry_convert.country_alpha2_to_country_name(artist_country)
    return country_name

  continent_name: |
    import pycountry_convert
    continent_code = pycountry_convert.country_alpha2_to_continent_code(artist_country)
    continent_name = pycountry_convert.convert_continent_code_to_continent_name(continent_code)
    return continent_name

paths:
  albumtype:album: Modern/$continent_name/$country_name/$albumartist/$album%aunique{}/$albumartist - $album - $track. $title

```

Error output when I do an import is:

```
beetsplug.inline.InlineError: error in inline path field code:
import pycountry_convert
continent_code = pycountry_convert.country_alpha2_to_continent_code(artist_country)
continent_name = pycountry_convert.convert_continent_code_to_continent_name(continent_code)
return continent_name

NameError: name 'artist_country' is not defined
```

**Comment by arsaboo** [comment]
Try:
```yaml
  country_name: |
    import pycountry_convert
    artist_country = item['artist_country']
    country_name = pycountry_convert.country_alpha2_to_country_name(artist_country)
    return country_name
```

**Reply by vazaha-nl** [reply]
Thanks for your answer, unfortunately this also doesn't work. I get `NameError: name 'item' is not defined`

**Comment by vazaha-nl** [comment]
I added some debug code to print all the available globals with

```
  country_name: |
    from pprint import pprint
    pprint(globals())
```

but `artist_country` is not available anywhere. There is a `items` var but the items also do not contain a `artist_country` property.

Also looking at the [documentation](https://beets.readthedocs.io/en/stable/dev/plugins.html#add-path-format-functions-and-fields)  it looks like the custom fields are only available in path definitions and not in other plugins? Then what I want is indeed not possible using the inline plugin, which is unfortunate. 

Maybe I will try to write my own plugin that adds `%country_name{country_code}` and `%continent_name{country_code}` path functions, that should accomplish what I want.
